gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -157,16 +186,29 @@
     // TODO: Clear feature markers when project is deactivated.
     // TODO: Since we depend on project stream from repo anyway, this transformation can be moved
     // into the repo?
+    // Features that are persisted to the local and remote dbs.
+    Flowable<ImmutableSet<MapFeature>> persistentFeatures =
+        projectRepository
+            .getActiveProject()
+            .switchMap(this::getFeaturesStream)
+            .map(this::toMapFeatures);
+    Flowable<ImmutableSet<MapFeature>> transientFeatures =
+        drawnPolylineVertices.map(
+            vertices ->
+                ImmutableSet.of(
+                    toMapPolygon(
+                        featureRepository.newPolygonFeature(
+                            selectedProject.getValue().get(),
+                            selectedLayer.getValue().get(),
+                            vertices))));
     this.mapFeatures =
         LiveDataReactiveStreams.fromPublisher(
             Flowable.combineLatest(
-                    projectRepository
-                        .getActiveProject()
-                        .switchMap(this::getFeaturesStream)
-                        .map(this::toMapFeatures),
-                    selectedFeature,
-                    this::updateSelectedFeature)
-                .distinctUntilChanged());
```

Comment:
Why is this removed?

Reply:
to keep polygon drawing UI updated.

Reply:
But the removed code is not added back anywhere. Have you validated that this is not a regression?

Reply:
added back.